drivers/usb/device/usb_dc_nrfx.c

```diff
@@ -1723,27 +1713,6 @@ int usb_dc_ep_write(const uint8_t ep, const uint8_t *const data,
 		return -EAGAIN;
 	}
 
-	/* NRFX driver performs the fragmentation if buffer length exceeds
-	 * maximum packet size, however in current implementation, data is
-	 * copied to the internal buffer and must me fragmented here.
-	 * In case of fragmentation, a flag is set to prevent triggering
-	 * status stage which is handled by hardware, because there will be
-	 * another write coming.
-	 */
-	if (data_len > ep_ctx->cfg.max_sz) {
```

Comment:
Do we need to check data_len for the iso IN endpoint?

Reply:
Yes. Buffer size depends on the configuration, the device is not allowed to send more than mps, the host would not accept it.